pkg/core/fetch.go

```diff
@@ -205,30 +200,13 @@ func Fetch(ctx context.Context, storage database.Storage, pm *plugin.Manager, op
 	}
 	// set metadata we want to pass to
 	metadata := map[string]interface{}{"cq_fetch_id": fetchId}
-	if opts.History != nil {
-		fd := opts.History.FetchDate()
-		log.Info().Str("fetch_date", fd.Format(time.RFC3339)).Stringer("fetch_id", fetchId).Msg("history enabled adding fetch date")
-		metadata["cq_fetch_date"] = fd
-		// TODO Remove(Compatibility): Code below is for providers using the old SDK version, where metadata isn't available in FetchRequest
-		// Removing this without updating provider will set cq_fetch_date to the time of execution start, which HistoryCfg.TimeTruncation doesn't apply
-		if opts.ExtraFields == nil {
-			opts.ExtraFields = make(map[string]interface{})
-		}
-		opts.ExtraFields["cq_fetch_date"] = fd
-	}
-	log.Info().Interface("extra_fields", opts.ExtraFields).Bool("history_enabled", opts.History != nil).Msg("received fetch request")
+	log.Info().Interface("extra_fields", opts.ExtraFields).Msg("received fetch request")
 
-	// TODO: in future more components will want state, so make state more generic and passable via database.Storage
-	db, err := sdkdb.New(ctx, logging.NewZHcLog(&log.Logger, "fetch"), storage.DSN())
+	stateClient, err := state.NewClient(ctx, storage.DSN())
```

Comment:
why not pass the state as part of the arguments, and create it once in the console.Client

Reply:
Done